### PR TITLE
features: equal

### PR DESCRIPTION
**Pull request by fenollp**
This begins the work of porting [a variant of] gogo's [equal plugin](https://github.com/gogo/protobuf/blob/b03c65ea87cdc3521ede29f62fe3ce239267c1bc/plugin/equal/equal.go)

An unrolled non-`reflect`ive implementation of `proto.Equal` is about twice faster in our testing.

Note: this is WIP. It is hard to test right now (I am working on a CI PR that should help here) mainly because this command cannot be used with forks:
```shell
go install github.com/planetscale/vtprotobuf/cmd/protoc-gen-go-vtproto@latest
```
Any help around this is very welcomed!

### CODE REVIEW

**Comment by vmg** [comment]
I think you're off to a good start but you're definitely going to need more tests for this feature!

>  It is hard to test right now (I am working on a CI PR that should help here) mainly because this command cannot be used with forks

Yeah, that's a limitation with `go install`. Your best bet is cloning your fork locally and then calling `go install` pointing at the local clone. That should get you going.

**Comment by fenollp** [comment]
@vmg Ready for review! :)

File: features/equal/equal.go [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+
+	for _, field := range message.Fields {
+		oneof := field.Oneof != nil && !field.Oneof.Desc.IsSynthetic()
+		nullable := field.Message != nil || (field.Oneof != nil && field.Oneof.Desc.IsSynthetic()) || (!proto3 && !oneof)
+		if !oneof {
+			p.field(field, nullable)
+		}
+	}
+
+	p.P(`return string(this.unknownFields) == string(that.unknownFields)`)
```
Comment (vmg) [inline features/equal/equal.go:99]: This should be `bytes.Equal`. It doesn't make much sense to force a `string` cast for comparison.

**Review by vmg — CHANGES_REQUESTED** [review]
This looks _great_. Thank you so much for the extensive tests. Could you please fix the `unknownFields` comparison?

**Review by vmg — APPROVED** [review]

**Pull request merged**